src/rpc/mining.cpp

```diff
@@ -157,10 +157,16 @@ UniValue generate(const JSONRPCRequest& request)
     UniValue blockHashes(UniValue::VARR);
     CReserveKey reservekey(pwalletMain);
     unsigned int nExtraNonce = 0;
+
+    std::vector<COutput> availableCoins;
+    if (fPoS && !pwalletMain->StakeableCoins(&availableCoins)) {
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "No available coins to stake");
+    }
```

Comment:
This should go inside the `while()` below.
Balance (and maturity) changes at every iteration. E.g. if we start before PoS activation, `availableCoins` is empty, and it never gets updated, so when it's time to stake block, it looks like we have no coins.

Reply:
yeah good, done.